tfhe/src/core_crypto/fft_impl/fft64/math/fft/x86.rs

```diff
@@ -78,21 +78,16 @@ pub fn mm256_cvtpd_epi64(simd: V3, x: __m256d) -> __m256i {
     avx2._mm256_blendv_epi8(value_if_positive, value_if_negative, sign_is_negative_mask)
 }
 
-/// Convert a vector of f64 values to a vector of i64 values.
-/// This intrinsics is currently not available in rust, so we have our own implementation using
-/// inline assembly.
-///
-/// The name matches Intel's convention (re-used by rust in their intrinsics) without the leading
-/// `_`.
-///
-/// [`Intel's documentation`](`https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm512_cvtt_roundpd_epi64 `)
+/// Convert a vector of f64 values to a vector of i64 values with rounding to nearest integer.
+/// [`Intel's documentation`](`https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm512_cvt_roundpd_epi64`)
 #[cfg(feature = "nightly-avx512")]
 #[inline(always)]
-pub fn mm512_cvtt_roundpd_epi64(simd: V4, x: __m512d) -> __m512i {
+pub fn mm512_cvt_round_nearest_pd_epi64(simd: V4, x: __m512d) -> __m512i {
+    let _ = simd.avx512dq;
+
     // SAFETY: simd contains an instance of avx512dq, that matches the target feature of
     // `implementation`
-    _ = simd;
-    unsafe { _mm512_cvttpd_epi64(x) }
+    unsafe { _mm512_cvt_roundpd_epi64::<{ _MM_FROUND_TO_NEAREST_INT | _MM_FROUND_NO_EXC }>(x) }
```

Comment:
should we round to nearest if we used to truncate ?

Reply:
we're not replacing truncate+convert by round+convert
we're replacing round+truncate+convert. the rounding was previously done in prologue_v4

Reply:
ah must have read too quick, let me check

Reply:
indeed thanks for the clarification